rd-net/Test.Lifetimes/Collections/Viewable/SequentialSchedulerTest.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Collections.Viewable;
+using JetBrains.Lifetimes;
+using NUnit.Framework;
+
+namespace Test.Lifetimes.Collections.Viewable
+{
+#if !NET35
+  [TestFixture(SchedulerKind.TaskScheduler)]
+  [TestFixture(SchedulerKind.ISchduler)]
+  public class SequentialSchedulerTest : LifetimesTestBase
+  {
+    private readonly SchedulerKind mySchedulerKind;
+
+    public SequentialSchedulerTest(SchedulerKind schedulerKind)
+    {
+      mySchedulerKind = schedulerKind;
+    }
+    
+    [Test]
+    public void SimpleTest()
+    {
+      const int n = 1000;
+      var scheduler = CreateSequentialScheduler("TestScheduler", TestLifetime);
+      
+      Assert.IsFalse(scheduler.OutOfOrderExecution, "scheduler.OutOfOrderExecution");
+      
+      {
+        Assert.IsFalse(scheduler.IsActive);
+        
+        for (var i = 0; i < 100; i++)
+        {
+          var sum = 0;
+          var range = Enumerable.Range(0, n);
+          var tasks = range.Select(j => TestLifetime.Start(scheduler, () =>
+          {
+            sum += j;
+            Assert.IsTrue(scheduler.IsActive);
+          })).ToArray();
+          
+          Task.WaitAll(tasks);
+          Assert.AreEqual(range.Sum(), sum);
+        }
+      }
+      
+      Assert.IsFalse(scheduler.IsActive);
+      
+      {
+        TestLifetime.StartAsync(scheduler, async () =>
+        {
+          Assert.IsTrue(scheduler.IsActive);
+          
+          var reached = false;
+          scheduler.Queue(() =>
+          {
+            reached = true;
+            Assert.IsTrue(scheduler.IsActive);
+          });
+          
+          SpinWait.SpinUntil(() => reached, TimeSpan.FromMilliseconds(100));
+          
+          Assert.IsFalse(reached);
```

Comment:
Here, I have the same comment about the `reached` variable as for the `hasValue` variable above. Probably it'd be better to avoid writing and reading an ordinary stack variable from different threads without any special measures applied.

Same for the `count` and `reached` variables below, in the next test.

Reply:
SequentialScheduler guarantees memory consistency.